payroll/views.py

```diff
@@ -198,35 +198,25 @@ def get_context_data(self, **kwargs):
         return context
 
     def aggregate_department_statistics(self):
-        query = '''
-            SELECT
-              employer.name,
-              AVG(salary.amount) AS average,
-              SUM(salary.amount) AS budget,
-              COUNT(*) AS headcount,
-              employer.slug AS slug
-            {from_clause}
-            WHERE employer.parent_id = {id}
-            GROUP BY employer.id, employer.name
-            ORDER BY SUM(salary.amount) DESC
-        '''.format(from_clause=self.from_clause,
-                   id=self.object.id)
-
-        total_expenditure = sum(self.object.employee_salaries)
-
-        with connection.cursor() as cursor:
-            cursor.execute(query)
+        entity_base_pay = Sum(Coalesce("positions__jobs__salaries__amount", 0))
+        entity_extra_pay = Sum(Coalesce("positions__jobs__salaries__extra_pay", 0))
+        median_total_pay = Percentile(
+            (
+                NullIf(
+                    Coalesce("positions__jobs__salaries__amount", 0) +
+                    Coalesce("positions__jobs__salaries__extra_pay", 0), 0
+                )
+            ), 0.5, output_field=FloatField()
+        )
 
-            department_salaries = []
-            for department, average, budget, headcount, slug in cursor:
-                department_salaries.append({
-                    'department': department,
-                    'amount': average,
-                    'total_budget': budget,
-                    'headcount': headcount,
-                    'slug': slug,
-                    'percentage': (float(budget / total_expenditure) * 100)  # Percentage of total expenditure
-                })
+        department_salaries = Employer.objects.filter(parent_id=self.object.id)\
+                                              .values("name", "slug")\
+                                              .annotate(headcount=Count("positions__jobs__salaries"),
+                                                        median_tp=median_total_pay,
+                                                        entity_bp=entity_base_pay,
+                                                        entity_ep=entity_extra_pay,
+                                                        total_expenditure=entity_base_pay + entity_extra_pay)\
+                                              .order_by("-total_expenditure")[::1]
```

Comment:
What is `[::1]` doing for you here?

Reply:
without slicing, this returns a queryset. According to [django docs](https://docs.djangoproject.com/en/3.0/ref/models/querysets/#when-querysets-are-evaluated), slicing is one way to force the evaluation of this queryset so that it returns a list. There were a couple of options I could have chosen; let me know if any of those is preferable to what I'm doing here. I could also add a comment

Reply:
Ah, I see! Is there any reason you can't return a queryset, instead of a list?

Reply:
Apparently not! looks like I slightly overthought that one.